### PR TITLE
doc: v2/en: added the 1.15.8.1 changelog.

### PR DESCRIPTION
**Pull request by thibaultcha**

### CODE REVIEW

File: v2/en/changelog-1015006.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+* resty-doc indexes: skipped nginx's njs docs due to the use of special xml tags.
+* upgraded [ngx_lua](https://github.com/openresty/lua-nginx-module#readme) to 0.10.14rc3.
+    * change: we no longer support the standard Lua 5.1 interpreter (PUC-Rio Lua). It is now strongly recommended to use the OpenResty LuaJIT releases, bundled with the official OpenResty releases.
+    * change: we now print an alert when a non openresty-specific version of LuaJIT is detected since many optimizations would be missing. Thanks Thibault Charbonnier for the patch.
+    * change: we now print a warning when writing to the Lua global variable _G.
+    * feature: added support for ARM64 (Aarch64). Thanks Cloudflare for sponsoring this work. Thanks Dejiang Zhu and Zexuan Luo for the development work of this patch.
+    * feature: implemented the receiveany() upstream cosocket method. Thanks spacewander for the patch.
+    * feature: implemented a new 'lua_sa_restart' directive, which sets the SA_RESTART flag on [nginx](nginx.html) workers signal dispositions. Thanks Thibault Charbonnier for the patch.
```
Comment (agentzh) [inline v2/en/changelog-1015006.md]: Should add a link to `lua_sa_restart`'s docs? Just update the resolve-links.pl script under `util/`.

File: v2/en/changelog-1015006.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+* upgraded the [nginx](nginx.html) core to 1.15.6.
+    * see the changes here: http://nginx.org/en/CHANGES
+* feature: ./configure: added new options --without-stream_ssl_module and --without-stream.
+* feature: ./configure: support @rpath placeholder in OS X. Thanks Datong Sun for the patch.
+* bugfix: [nginx](nginx.html) did not destroy the cycle memory pool before the daemon process exits. Thanks Yuansheng Wang for the patch.
+* resty-doc indexes: skipped nginx's njs docs due to the use of special xml tags.
+* upgraded [ngx_lua](https://github.com/openresty/lua-nginx-module#readme) to 0.10.14rc3.
+    * change: we no longer support the standard Lua 5.1 interpreter (PUC-Rio Lua). It is now strongly recommended to use the OpenResty LuaJIT releases, bundled with the official OpenResty releases.
+    * change: we now print an alert when a non openresty-specific version of LuaJIT is detected since many optimizations would be missing. Thanks Thibault Charbonnier for the patch.
+    * change: we now print a warning when writing to the Lua global variable _G.
```
Comment (agentzh) [inline v2/en/changelog-1015006.md]: Better add backticks around `_G`.

File: v2/en/changelog-1015006.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    * feature: added support for ARM64 (Aarch64). Thanks Cloudflare for sponsoring this work. Thanks Dejiang Zhu and Zexuan Luo for the development work of this patch.
+    * feature: implemented the receiveany() upstream cosocket method. Thanks spacewander for the patch.
```
Comment (agentzh) [inline v2/en/changelog-1015006.md]: Better add a link to the `receiveany()`'s docs?

File: v2/en/changelog-1015006.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    * bugfix: fixed segfault in [NGINX](nginx.html) core >= 1.15.0 when [init_worker_by_lua*](https://github.com/openresty/lua-nginx-module#init_worker_by_lua) is used. Thanks Datong Sun for the patch.
+    * bugfix: [ngx.process](https://github.com/openresty/lua-resty-core/blob/master/lib/ngx/process.md#readme): [type()](https://github.com/openresty/lua-resty-core/blob/master/lib/ngx/process.md#type) didn't return 'master' in master process. Thanks spacewander for the patch.
+    * bugfix: [tcpsock:setkeepalive](https://github.com/openresty/lua-nginx-module#tcpsocksetkeepalive): worker processes might take too long to gracefully shut down when the keep alive connections take a long max idle time. Thanks Dejiang Zhu for the patch.
+    * bugfix: when variable is cacheable and indexed, setting the request "Host" header should clear any existing cached $host variable value so that subsequent reads return the expected result.
+    * bugfix: silenced -Wcast-function-type warnings, allowing for compilation with gcc8. Thanks Alessandro Ghedini for the patch.
```
Comment (agentzh) [inline v2/en/changelog-1015006.md]: Better ad backticks around `-Wxxx`.

File: v2/en/changelog-1015006.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+    * bugfix: the 'dd' calls in ngx_http_lua_util.h are now guarded by '#ifdef DDEBUG'. Thanks spacewander for the patch.
+    * optimize: removed a duplicated include directive in ngx_http_lua_regex.c. Thanks Thibault Charbonnier for the patch.
```
Comment (agentzh) [inline v2/en/changelog-1015006.md]: This is too trivial to mention. Better remove other similarly trivial entries (if any).

File: v2/en/changelog-1015006.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+    * doc: mention that [ngx.req](https://github.com/openresty/lua-resty-core/blob/master/lib/ngx/req.md#readme).set_body_data() and [ngx.req](https://github.com/openresty/lua-resty-core/blob/master/lib/ngx/req.md#readme).set_body_file() must read the request body. Thanks Thibault Charbonnier for the patch.
+    * doc: fixed the links to ssl_session_store_by_lua_block. Thanks chronolow for the patch.
+    * typo: fixed a debug log in access and rewrite handlers. Thanks sbhr for the patch.
+* upgraded [ngx_stream_lua](https://github.com/openresty/stream-lua-nginx-module#readme) to 0.0.6rc3.
+    * change: we now print an alert when a non openresty-specific version of LuaJIT is detected since many optimizations would be missing. Thanks Thibault Charbonnier for the patch.
+    * feature: added support for ARM64. Thanks Cloudflare for sponsoring this work. Thanks Dejiang Zhu and Zexuan Luo for the development work of this patch.
+    * feature: ssl_certby_lua* support. Thanks Kong Inc. for sponsoring the development of this feature, and thanks James Callahan and Datong Sun for the development work of this patch.
```
Comment (agentzh) [inline v2/en/changelog-1015006.md]: Should use `ssl_certificate_by_lua*` instead. Also needs a doc link.

File: v2/en/changelog-1015006.md [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+    * bugfix: the 'dd' calls in ngx_http_lua_util.h are now guarded by '#ifdef DDEBUG'. Thanks spacewander for the patch.
+    * optimize: removed a duplicated include directive in ngx_http_lua_regex.c. Thanks Thibault Charbonnier for the patch.
+    * misc: updated _G write guard message to be more accurate. Thanks Thibault Charbonnier for the patch.
+    * misc: rename a test file to be properly ordered. Thanks Thibault Charbonnier for the patch.
+    * doc: updated the docs to reflect the change that we now no longer support the standard Lua 5.1 interpreter in this module. also recommended OpenResty's LuaJIT branch version instead of the stock LuaJIT.
+    * doc: mention that [ngx.req](https://github.com/openresty/lua-resty-core/blob/master/lib/ngx/req.md#readme).set_body_data() and [ngx.req](https://github.com/openresty/lua-resty-core/blob/master/lib/ngx/req.md#readme).set_body_file() must read the request body. Thanks Thibault Charbonnier for the patch.
+    * doc: fixed the links to ssl_session_store_by_lua_block. Thanks chronolow for the patch.
```
Comment (agentzh) [inline v2/en/changelog-1015006.md]: Needs a doc link.

File: v2/en/changelog-1015006.md [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+    * feature: shdict: enabled the FFI-based API for the stream subsystem. Thanks Datong Sun for the patch.
+    * feature: [ngx.semaphore](https://github.com/openresty/lua-resty-core/blob/master/lib/ngx/semaphore.md#readme): enabled the FFI-based API for the stream subsystem. Thanks Datong Sun for the patch.
+    * feature: errlog: enabled the FFI-based API for the stream subsystem. Thanks Thibault Charbonnier for the patch.
+    * feature: also implemented the ndk.* API with [FFI](http://luajit.org/ext_ffi.html).
+    * bugfix: [ngx.process](https://github.com/openresty/lua-resty-core/blob/master/lib/ngx/process.md#readme): [type()](https://github.com/openresty/lua-resty-core/blob/master/lib/ngx/process.md#type) didn't return 'master' in master process. Thanks spacewander for the patch.
+    * bugfix: now we only allow a single version of [ngx_lua](https://github.com/openresty/lua-nginx-module#readme) and [ngx_stream_lua](https://github.com/openresty/stream-lua-nginx-module#readme). Thanks spacewander for the patch.
+    * optimize: removed unnecessary ffi declarations of struct ngx_http_request_s from resty.core.base. Thanks spacewander for the patch.
```
Comment (agentzh) [inline v2/en/changelog-1015006.md]: This is too trivial to mention.

File: v2/en/changelog-1015006.md [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+    * doc: [ngx.process](https://github.com/openresty/lua-resty-core/blob/master/lib/ngx/process.md#readme): updated the process type list. Thanks spacewander for the patch.
+    * doc: [ngx.balancer](https://github.com/openresty/lua-resty-core/blob/master/lib/ngx/balancer.md#readme): fixed a typo in the sample code. Thanks chronolaw for the patch.
+    * doc: [ngx.ssl](https://github.com/openresty/lua-resty-core/blob/master/lib/ngx/ssl.md#readme): documented the get
+    * doc: [ngx.semaphore](https://github.com/openresty/lua-resty-core/blob/master/lib/ngx/semaphore.md#readme): switch status to production ready.
+* upgraded LuaJIT to 2.1-20181029: https://github.com/openresty/luajit2/tags
+    * feature: implemented the new Lua and C API functions for thread exdata. This API is used internally by the OpenResty and we advise not to use it yourself in the context of OpenResty. Thanks Zexuan Luo for preparing the final version of the patch.
+    * feature: luajit.h: defined the macro OPENRESTY_LUAJIT for our branch of LuaJIT.
```
Comment (agentzh) [inline v2/en/changelog-1015006.md]: Needs backticks around identifiers like `OPENRESTY_LUAJIT`.

File: v2/en/changelog-1015008.md [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+    * change: we now print an alert when a non openresty-specific version of LuaJIT is detected since many optimizations would be missing. Thanks Thibault Charbonnier for the patch.
+    * change: we now print a warning when writing to the Lua global variable `_G`.
+    * change: we now avoid running [init_by_lua*](https://github.com/openresty/lua-nginx-module#init_by_lua) in signaller processes and when testing the [nginx](nginx.html) configuration. Thanks spacewander for the patch.
+    * change: we now avoid generating the Content-Type response header when getting all response headers. Thanks spacewander for the patch.
+    * feature: added support for ARM64 (Aarch64). Thanks Cloudflare for sponsoring this work. Thanks Dejiang Zhu and Zexuan Luo for the development work of this patch.
+    * feature: added the C API to support the [ngx.pipe](https://github.com/openresty/lua-resty-core/blob/master/lib/ngx/pipe.md#readme) API provided by [lua-resty-core](https://github.com/openresty/lua-resty-core#readme). Thanks spacewander for the patch.
+    * feature: added backlog queueing to cosocket connection pools. Thanks spacewander for the patch.
```
Comment (agentzh) [inline v2/en/changelog-1015008.md]: Better add a link to the corresponding documentation section? This is a major feature :)

File: v2/en/changelog-1015008.md [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+# Version 1.15.8.1 - dd MMMM YYYY
+
+* upgraded the [nginx](nginx.html) core to 1.15.8.
+    * see the changes here: http://nginx.org/en/CHANGES
+* change: we now enable the GC64 mode by default in our bundled LuaJIT build for x86_64 architectures; this can be disabled with ./configure `--without-luajit-gc64`. Thanks Thibault Charbonnier for the patch.
+* feature: bundled the [lua-tablepool](https://github.com/openresty/lua-tablepool#readme) library. Thanks Thibault Charbonnier for the patch.
+* feature: ./configure: added new options `--without-stream_ssl_module` and `--without-stream`.
+* feature: ./configure: added support for the `-h` option.
+* feature: ./configure: support `@rpath` placeholder in OS X. Thanks Datong Sun for the patch.
+* feature: updated the socket_cloexec patches to support the [ngx.pipe](https://github.com/openresty/lua-resty-core/blob/master/lib/ngx/pipe.md#readme) API. Thanks spacewander for the patch.
```
Comment (agentzh) [inline v2/en/changelog-1015008.md]: Better add backticks around `socket_cloexec`?

File: v2/en/changelog-1015008.md (hunk 10 above)
Comment (thibaultcha) [reply v2/en/changelog-1015008.md]: Good point!

File: v2/en/changelog-1015008.md [hunk 12]
```diff
@@ -0,0 +1,124 @@
+<!---
+    @title         ChangeLog 1.15.8
+    @creator       Thibault Charbonnier
+    @created       2018-11-14 01:32 GMT
+--->
+
+# Version 1.15.8.1 - dd MM YYYY
+
+* upgraded the [nginx](nginx.html) core to 1.15.8.
```
Comment (thibaultcha) [inline v2/en/changelog-1015008.md]: need to revisit the nginx links here, something went wrong.

**Comment by agentzh** [comment]
Already merged.

**Pull request closed without merging**